sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -15,11 +15,23 @@ use crate::transport::{ProxyOps, Transport, TransportError};
 use anyhow::Result;
 
 use erased_serde::Serialize;
+use indicatif::{ProgressBar, ProgressStyle};
 use std::any::Any;
 use std::cell::RefCell;
 use std::collections::HashMap;
 use std::rc::Rc;
 
+/// Helper function to create a progress bar in the same form for each of
+/// the commands which will use it.
+pub fn progress_bar(total: u64) -> ProgressBar {
+    let progress = ProgressBar::new(total);
+    progress.set_style(
+        ProgressStyle::default_bar()
+            .template("[{elapsed_precise}] [{wide_bar}] {bytes}/{total_bytes} ({eta})"),
+    );
+    progress
+}
```

Comment:
Unless I'm missing something, `progress_bar` is only used in opentitantool, not opentitan lib. Why was it moved to opentitanlib? Is there a reason why users of opentitanlib other than opentitantool would want to use the same progress bar configuration as opentitantool?

Reply:
Yes: it was moved to opentitanlib::app, a collection of useful things you might want when building an application on top of opentitanlib.

This currently includes the `CommandDispatch` trait, the `TransportWrapper` struct and the pinmap/etc configuration structs.  This is probably in need of a refactor for better separation of concerns.

Reply:
This could also be useful for e2e tests (if we have any tests that will take some time).